promql/engine.go

```diff
@@ -622,14 +622,18 @@ func (ng *Engine) populateSeries(querier storage.Querier, s *parser.EvalStmt) {
 	// The evaluation of the VectorSelector inside then evaluates the given range and unsets
 	// the variable.
 	var evalRange time.Duration
+	// Whenever a SubqueryExpr is evaluated, subqueryStep is set to the corresponding step.
+	// The evaluation of the VectorSelector inside then uses this.
+	var subqueryStep time.Duration
```

Comment:
This is a local over the whole expression, so if you had somethging like `max_over_time(a[1s:1s]) + a` then we'd end up incorrectly sending a subquery step with the 2nd selector.